app/login/login-controller.coffee

```diff
@@ -2,55 +2,32 @@
 
 class Controller
     
-  constructor: (@authService) ->
+  constructor: (@authService, @profileService, @messageService, @state) ->
     @user = {
       email: ''
       password: ''
     }
     @registerCheck = false
-    @submitted = false
-    @error = ''
-    @success = ''
-    @messageError = false
-    @messageSuccess = false
+    @loginStatus = @messageService.newEv()
     
   submit: ->
-    @submitted = true
-    @error = ''
-    @success = ''
-    @messageError = false
-    @messageSuccess = false
+    @loginStatus.start()
     if @registerCheck
       @register()
     else
       @login()
-        
-  parseError: (error) ->
-    switch error
-      when 'INVALID_EMAIL'
-        return 'The specified user account email is invalid.'
-      when 'INVALID_PASSWORD'
-        return 'The specified user account password is incorrect.'
-      when 'INVALID_USER'
-        return 'The specified user account does not exist.'
-      when 'EMAIL_TAKEN'
-        return 'The specified user account email is already taken.'
-      else
-        return 'Error logging user in.'
+    return
         
   register: ->
     @authService.register(@user)
-      .then(=>
-        @submitted = false
-        @success = 'Registered successfully'
-        @messageSuccess = true
+      .then((authData)=>
+        @loginStatus.stopOk()
+        @profileService.createUser(authData.uid)
         @login()
         return
       )
       .catch((error) =>
-        @submitted = false
-        @error = @parseError(error.code)
-        @messageError = true
+        @loginStatus.stopKo(@authService.parseError(error.code))
         return
```

Comment:
never use explicit return use undefined instead
try to avoid anonymous functions are more difficult to debug and trace the errors